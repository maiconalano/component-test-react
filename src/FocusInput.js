import React, { useRef } from "react";

const FocusInput = () => {
  const inputRef = useRef(null);
  return (
    <div>
      <input
        type="text"
        aria-label="focus-input"
        ref={inputRef}
        placeholder="Foco aqui!!!"
      />
      <button onClick={() => inputRef.current.focus()}>
        Clique aqui para focar
      </button>
    </div>
  );
};

export default FocusInput;
