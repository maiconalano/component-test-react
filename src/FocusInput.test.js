import React from "react";
import FocusInput from "../src/FocusInput";
import { render, fireEvent, cleanup } from "@testing-library/react";

afterEach(cleanup);

test("Validar FocusInput snapshot", () => {
  const { container } = render(<FocusInput />);
  expect(container.firstChild).toMatchSnapshot();
});

test("Validar clique para focar", () => {
  const { getByPlaceholderText, getByText } = render(<FocusInput />);
  fireEvent.click(getByText("Clique aqui para focar"));
  expect(document.activeElement).toBe(getByPlaceholderText("Foco aqui!!!"));
});
