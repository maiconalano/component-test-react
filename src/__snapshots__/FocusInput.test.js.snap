// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar FocusInput snapshot 1`] = `
<div>
  <input
    aria-label="focus-input"
    placeholder="Foco aqui!!!"
    type="text"
  />
  <button>
    Clique aqui para focar
  </button>
</div>
`;
